src/components/OntologyBrowser/store.ts

```diff
@@ -212,6 +123,32 @@ export class PhenotypeStore {
     }
   };
 
+  getTreeNodeForKey = (key: string, treeNode = this.tree): TreeNode | null => {
+    let result: TreeNode | null = null;
+    for (let i = 0; i < treeNode.length; i++) {
+      if (treeNode[i].key === key) {
+        result = treeNode[i];
+        break;
+      }
+      result = this.getTreeNodeForKey(key, treeNode[i].children);
+      if (result) {
+        break;
+      }
+    }
+    return result;
+  };
+
+  getChildrenKeys = (node: TreeNode, root = false) => {
+    let nKeys: string[] = [];
+    node.children.forEach((i) => {
+      nKeys = nKeys.concat(this.getChildrenKeys(i));
+    });
+    if (!root) {
+      nKeys.push(node.key);
```

Comment:
is using a more functional style (like using map) I would do this [...nKeys, node.key].But it s just a matter of taste

Reply:
this does not address that the mutation/concatenation is conditional